### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by MiaoJieF** [lly00412/SEDNet#3]
SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.

1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?

**Comment by MiaoJieF** [comment]
Could you please answer or recommend some relevant papers? I am very interested in this problem.

**Comment by If-only1** [comment]
> SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.
> 
> 1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?

I'm not the author of the original paper, I just happened to see this question.
From some of my recent UNCERTAINTY experiments, UNCERTAINTY loss does not significantly improve the accuracy of the model in the case of 100%DENSE prediction, and even decreases it, unless there is some noise in the supervised signal.

**Comment by MiaoJieF** [comment]
> 

My recent experiments came to the same conclusion as yours.

**Comment by MiaoJieF** [comment]
> > SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.
> > 
> > 1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?
> 
> I'm not the author of the original paper, I just happened to see this question. From some of my recent UNCERTAINTY experiments, UNCERTAINTY loss does not significantly improve the accuracy of the model in the case of 100%DENSE prediction, and even decreases it, unless there is some noise in the supervised signal.

My recent experiments came to the same conclusion as yours.

**Comment by lly00412** [comment]
> SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.
> 
> 1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?

Sorry for replying late. 
The idea is to improve not only the accuracy of the disparity estimation but also the accuracy of uncertainty estimation, i.e. make the uncertainty estimates match the errors.  However, we do not have the ground truth of the uncertainty per pixel, hence we try to match in distribution level.  A similar idea can be found in [(Bailey et al.)(https://ieeexplore.ieee.org/iel5/4058334/4058335/04058955.pdf?casa_token=nYs66YHYjVIAAAAA:pKIZ-frahjBIooZUVH4kAis9foez0TXJmgt7TLFqNU97lhcizhJeDdms1Go4WWiWDxSdCxw3Ie0)] 
Based on the experiments we did so far, we did not observe overfitting on the training set, but observed mismatching between uncertainty and disparity errors, please refer to Figure S.2 ~ S.8 in the supplement of our paper.

**Comment by MiaoJieF** [comment]
> > SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.
> > 
> > 1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?
> 
> Sorry for replying late. The idea is to improve not only the accuracy of the disparity estimation but also the accuracy of uncertainty estimation, i.e. make the uncertainty estimates match the errors. However, we do not have the ground truth of the uncertainty per pixel, hence we try to match in distribution level. A similar idea can be found in [(Bailey et al.)(https://ieeexplore.ieee.org/iel5/4058334/4058335/04058955.pdf?casa_token=nYs66YHYjVIAAAAA:pKIZ-frahjBIooZUVH4kAis9foez0TXJmgt7TLFqNU97lhcizhJeDdms1Go4WWiWDxSdCxw3Ie0)] Based on the experiments we did so far, we did not observe overfitting on the training set, but observed mismatching between uncertainty and disparity errors, please refer to Figure S.2 ~ S.8 in the supplement of our paper.

Thank you for your reply!

I still can't understand, if GT is dense and accurate (VKITTI used in the paper is a virtual dataset, disparity is dense), why do you need another loss function? The experimental results in the paper seem to show that using UNCERTAINTY loss can improve EPE, which confuses me.

I am trying to apply your method to other tasks and found that adding UNCERTAINTY loss does not improve the accuracy. This frustrates me.

**Comment by lly00412** [comment]
> > > SEDNet is an excellent work. After reading the paper and the code, I have some questions and hope to get your help.
> > > 
> > > 1. Why does adding an extra uncertainty loss improve performance on in-domain data? Isn't it enough to have ground truth to supervise? Could it be that the network is overfitting on the training set?
> > 
> > 
> > Sorry for replying late. The idea is to improve not only the accuracy of the disparity estimation but also the accuracy of uncertainty estimation, i.e. make the uncertainty estimates match the errors. However, we do not have the ground truth of the uncertainty per pixel, hence we try to match in distribution level. A similar idea can be found in [(Bailey et al.)(https://ieeexplore.ieee.org/iel5/4058334/4058335/04058955.pdf?casa_token=nYs66YHYjVIAAAAA:pKIZ-frahjBIooZUVH4kAis9foez0TXJmgt7TLFqNU97lhcizhJeDdms1Go4WWiWDxSdCxw3Ie0)] Based on the experiments we did so far, we did not observe overfitting on the training set, but observed mismatching between uncertainty and disparity errors, please refer to Figure S.2 ~ S.8 in the supplement of our paper.
> 
> Thank you for your reply!
> 
> I still can't understand, if GT is dense and accurate (VKITTI used in the paper is a virtual dataset, disparity is dense), why do you need another loss function? The experimental results in the paper seem to show that using UNCERTAINTY loss can improve EPE, which confuses me.
> 
> I am trying to apply your method to other tasks and found that adding UNCERTAINTY loss does not improve the accuracy. This frustrates me.

The algorithm is designed to support the multi-task framework. In most of the real-world problems, it is actually hard to have dense ground truth. Many papers have proved that uncertainty estimation benefits CV tasks. I recommend you read papers about uncertainty estimation theory, which may help you understand this problem better. This repository is only used to maintain the code for SEDNet.
